src/com/goide/highlighting/GoAnnotator.java

```diff
@@ -110,6 +110,47 @@ else if (element instanceof GoBuiltinCallExpr) {
         checkMakeCall(call, holder);
       }
     }
+    else if (element instanceof GoFunctionLit) {
+      checkLiteralEvaluatedButNotUsed(holder, element, "Function");
+    }
+    else if (element instanceof GoStringLiteral) {
+      checkLiteralEvaluatedButNotUsed(holder, element, "String");
+    }
+    else if (element instanceof GoLiteral) {
+      String message = ((GoLiteral)element).getChar() != null ? "Rune" : "Number";
+      checkLiteralEvaluatedButNotUsed(holder, element, message);
+    }
+  }
+
+  private static void checkLiteralEvaluatedButNotUsed(@NotNull AnnotationHolder holder, @NotNull PsiElement element, String message) {
+    PsiElement parent = element.getParent();
+    if (element instanceof GoFunctionLit) {
+      if (parent instanceof GoCallExpr) return;
+    }
+    else if (element instanceof GoStringLiteral ||
+             element instanceof GoLiteral) {
+      if (parent instanceof GoIndexOrSliceExpr ||
+          parent instanceof GoConditionalExpr) {
+        return;
+      }
+    }
+
+    parent = PsiTreeUtil.getParentOfType(element, GoLeftHandExprList.class, GoArgumentList.class);
+    if (parent == null) return;
+    if (parent instanceof GoArgumentList) return;
+    if (parent instanceof GoLeftHandExprList &&
+        ((GoLeftHandExprList)parent).getExpressionList().size() != 1) {
+      return;
+    }
+
+    if (element instanceof GoFunctionLit) {
+      element = ((GoFunctionLit)element).getFunc();
+    }
+    else if (parent.getParent().getParent() instanceof GoSwitchStatement) {
```

Comment:
No false positives as far as I can see on my GOPATH / Go SDK.
While inspecting the Go SDK however I've got: https://ea.jetbrains.com/browser/ea_reports/1010007 but I'm not sure if it's related or not.
Also, it does take quite a bit to perform all the inspections as it seems, do you think I should move this one up to it's own inspection?